tests/arguments_passing_regression/any_success_all_fail_new.py

```diff
@@ -0,0 +1,30 @@
+from hera.workflows import DAG, Workflow, script
```

Comment:
Tbf the number of tests does help instill confidence, but I'm wondering if it would be easier/clearer to have a select few regression tests that use certain features, e.g. `with_param`. I feel like we're bloating the tests with regression tests on all the examples vs targeted tests

Reply:
Maybe the regression tests should be a part of a separate test suite that only runs on PRs, not on `make test`?